tests/installation/bootloader_uefi.pm

```diff
@@ -89,6 +90,16 @@ sub run {
     # so 'regurl' does not apply there.
     registration_bootloader_params(utils::VERY_SLOW_TYPING_SPEED) unless (is_jeos or is_caasp);
 
+    # if a support_server is used, we need to wait for him to finish its initialization
+    # and we need to do it *before* starting the OS, as a DHCP request can happen too early
+    if (check_var('USE_SUPPORT_SERVER', 1)) {
+        diag "Waiting for support server to complete setup...";
+
+        # we use mutex to do this
+        mutex_lock('support_server_ready');
+        mutex_unlock('support_server_ready');
```

Comment:
I wonder if there isn't a better way than this "lock/unlock" tuples

Reply:
I try to find another way but failed. But I'm not an openQA expert :)
We need to wait for a 'signal' shared between two or more VM. Barriers exists too, if you think it's better I can try to use it instead.

Reply:
But the main issue with barriers is that a number of 'waiters' need to be assign when calling barrier_create. And in the case cover by the PR we don't know the number of 'waiters'.